src/measurement_buffer.cpp

```diff
@@ -152,7 +152,13 @@ void MeasurementBuffer::BufferPCLCloud(const \
 
     point_cloud_ptr cld_global(new pcl::PointCloud<pcl::PointXYZ>);
 
-    _buffer.transform(cloud, *cld_global, _global_frame);
+    // Create a tf::Transform using the tfStamped for _buffer
+    geometry_msgs::TransformStamped tf_stamped = _buffer.lookupTransform(_global_frame, cloud.header.frame_id.c_str(), ros::Time::now(), ros::Duration(5.0));
```

Comment:
Very last thing: you have the timeout to 5 seconds, that's a huge amount, it should be something like 0.1 seconds that sounds more reasonable. 

Can you give me the :+1: that this is all working now? Will merge

Reply:
It is working !
Here are 2 videos of the layer working on our robot : 
- https://youtu.be/Ziaj99DUJgo
- https://youtu.be/71uVkSqQ3G8

The slow update is most likely due to our platform limitation